internal/db/schema/migrations/oss/postgres/56/01_session_connection_bytes_up_down_trigger.up.sql

```diff
@@ -0,0 +1,23 @@
+begin;
+
+create or replace function prevent_bytes_update_on_closed_connection() returns trigger
+as $$
+begin
+  if old.closed_reason is not null then
+    -- ignore new bytes_up and bytes_down
+    new.bytes_up := old.bytes_up;
+    new.bytes_down := old.bytes_down;
+  end if;
+  return new; -- any other fields are updated
+end;
+$$ language plpgsql;
+comment on function prevent_bytes_update_on_closed_connection is
+  'The last update of bytes_up and bytes_down for any session_connection should
+  be the one that happens as the consequence of a connection closure
+  (when closed_reason is also set). This function ensures those fields cannot
+  be updated past connection closure';
+
+create trigger update_connection_bytes before update of bytes_up, bytes_down on session_connection
+  for each row execute procedure prevent_bytes_update_on_closed_connection();
+
+commit;
```

Comment:
Do we want this? 

Introducing this check in the database acts as a second layer of defense against potentially bad state in the db (the first being the [update query itself](https://github.com/hashicorp/boundary/pull/2503/files#diff-249aec8bcdb2e6880afd550f5cdb35d4026fedf391dd5425f600babc63bf3438R107)), however it can also be confusing from a design perspective as to why we carved out those two fields specifically, considering that no field should really be updated after a connection is marked as closed.

cc @mgaffney @tmessi

Reply:
I don't like this. It feels like we are moving domain logic into the database.

Reply:
Addressed in [ac9b087](https://github.com/hashicorp/boundary/pull/2503/commits/ac9b087b0c553e3516db4bc54a0dbd7bbdbc7c6b)